src/lib/knowledge.ts

```diff
@@ -29,21 +28,26 @@ export const loadKnowledge = (id: string): Knowledge => {
 	return knowledge;
 };
 
-export const saveKnowledge = (Knowledge: Knowledge): void => {
+export const saveKnowledge = (knowledge: Knowledge): void => {
 	// Retrieve the current knowledges
 	const currentKnowledges = get(knowledgeStore) || [];
 
 	// Find the index of the knowledge with the same id, if it exists
-	const existingKnowledge = currentKnowledges.findIndex(s => s.id === Knowledge.id);
+	const existingIndex = currentKnowledges.findIndex((k) => k.id === knowledge.id);
 
-	if (existingKnowledge !== -1) {
+	if (existingIndex !== -1) {
 		// Update the existing knowledge
-		currentKnowledges[existingKnowledge] = Knowledge;
+		currentKnowledges[existingIndex] = knowledge;
 	} else {
 		// Add the new knowledge if it doesn't exist
-		currentKnowledges.push(Knowledge);
+		currentKnowledges.push(knowledge);
```

Comment:
Thanks for cleaning up these definitions, the variable names make more sense now 👍